### PR TITLE
Implements Default half Functionality #52

### PR DESCRIPTION
**Pull request by bharathgs**

### CODE REVIEW

**Comment by AkashGutha** [comment]
is there no inplace version for this one?

**Comment by bharathgs** [comment]
In line with [this](http://pytorch.org/docs/master/tensors.html#torch.Tensor.half) and #52 . We do not have an *in_place* version of `half()`.

**Comment by aiorla** [comment]
I like it @bharathgs :+1:. Could you change the inline doc to be like #273?

**Comment by bharathgs** [comment]
@aiorla thanks for pointing out that. We should adhere to this strictly.

File: syft/tensor.py [hunk 1]
```diff
@@ -1348,7 +1348,16 @@ def mm(self, tensor2):
         return syft.mm(self, tensor2)
 
     def half(self):
-        """cast the tensor to half-precision float type."""
+                """
```
Comment (AkashGutha) [inline syft/tensor.py]: spacing needs dome fix here.

Comment (AkashGutha) [reply syft/tensor.py]: ignore this. The issue was fixed. My lazy eye missed the last commit.

File: syft/tensor.py [hunk 2]
```diff
@@ -1353,6 +1353,23 @@ def mm(self, tensor2):
 
         return syft.mm(self, tensor2)
 
+    def half(self):
+        """
+        casts the tensor to half-precision float type
+
+        ----------
```
Comment (AkashGutha) [inline syft/tensor.py]: There are more separators than required. please check that

Comment (AkashGutha) [reply syft/tensor.py]: this thing is also in the numpy example. not sure why.

File: syft/tensor.py [hunk 3]
```diff
@@ -1397,6 +1397,23 @@ def mm(self, tensor2):
...
```
Comment (siarez) [inline syft/tensor.py]: To adhere to the docstring format in #252, it should say 
```
Parameters
----------
```
even when there are no parameters

**Pull request merged**